Edits and updates to 'Creating' and 'Template-based Setup' chapters.

rchrdgrace opened the pull request:
Some corrections and updates, particularly to Creating Azure Cloud Services for XT chapter. Found a number of minor dependencies that can possibly trip new users up.

ghost commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/microsoft/ExperimentTools?pullRequest=4) <br/>All CLA requirements met.

rchrdgrace commented:
Also committed a significant update for two important chapters: the "Integrating XT..." chapter, which describes the Code settings to undertake and some Config File information. I've added a lot of new cross-references to subsections in the xt_config_file chapter, which I think will significantly help out the readers who need to configure these things but might otherwise get confused or lose a lot of time fumbling for targeted information. Also corrected some formatting errors across both files

Changes are all on XTDOCSPREV for easy review.

Next set of changes for other files will be in a separate PR.

jlema commented:
Closing this PR as all commits are included in #5.
